packages/analytics/index.ts

```diff
@@ -59,7 +60,7 @@ export function registerAnalytics(instance: _FirebaseNamespace): void {
             .getImmediate();
           return {
             logEvent: analytics.logEvent
-          };
+          } as FirebaseAnalyticsInternal;
```

Comment:
I think you can assign the factory method to a const before calling `registerComponent`, and set the type to `InstanceFactory` there instead of casting.

Reply:
I removed it and there is no error. It's probably something I added before the type safety changes.